### DISCUSSION TITLE
Question (saving content as plain text, recrawling)

### DISCUSSION THREAD

**Discussion by zhaobm** [internetarchive/heritrix3#525] [Q&A] [answered]
Question 1: Can the crawled data be saved locally as text?

Question 2: After a link is climbed, can it be climbed again?

Current test results, the same link is crawling again error
bug is： 
           严重: Can't create bean 'warcWriter'; Can't create bean 'dispositionProcessors'; Can't create bean 'crawlController'; Can't create bean 'frontier'; Can't create bean 'seeds'
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'seeds': Unsatisfied dependency expressed through method 'setSeedListeners' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'frontier': Unsatisfied dependency expressed through method 'setCrawlController' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'crawlController': Unsatisfied dependency expressed through method 'setDispositionChain' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dispositionProcessors' defined in URL [file:/D:/MyProduct/heritrix3/./jobs/spider2/crawler-beans.cxml]: Cannot resolve reference to bean 'warcWriter' while setting bean property 'processors' with key [0]; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'warcWriter' defined in URL [file:/D:/MyProduct/heritrix3/./jobs/spider2/crawler-beans.cxml]: Instantiation of bean failed; nested exception is java.lang.ExceptionInInitializerError

**Comment by ato** [comment, marked as answer]
1. Heritrix doesn't currently have a builtin way to extract text from HTML pages.
2. My understanding is the easiest way to recrawl pages is to run the job again (with [deduplication configured](https://github.com/internetarchive/heritrix3/wiki/Duplication%20Reduction%20Processors) if desired). There's some builtin support for continuous recrawling mentioned in the wiki but I'm not sure if it was ever completed or how functional it is.
3. This sounds like a configuration error but without more details I don't know what the problem is.
